MobiFlight/MobiFlightFirmwareUpdater.cs

```diff
@@ -38,31 +39,108 @@ public static bool Update(MobiFlightModule module)
             String Port = module.InitUploadAndReturnUploadPort();
             if (module.Connected) module.Disconnect();
 
-            while (!SerialPort.GetPortNames().Contains(Port))
-            {
-                System.Threading.Thread.Sleep(100);
-            }
-
             if (module.Board.AvrDudeSettings != null)
             {
-                try {
+                while (!SerialPort.GetPortNames().Contains(Port))
+                {
+                    System.Threading.Thread.Sleep(100);
+                }
+
+                try
+                {
                     RunAvrDude(Port, module.Board);
                     result = true;
-                } catch(Exception e) {
+                } catch (Exception e) {
                     result = false;
                 }
 
                 if (module.Board.Connection.DelayAfterFirmwareUpdate > 0)
                 {
                     System.Threading.Thread.Sleep(module.Board.Connection.DelayAfterFirmwareUpdate);
                 }
-            } else
+            }
+            else if (module.Board.UsbDriveSettings != null)
+            {
+                try 
+                { 
+                    FlashViaUsbDrive(module.Board);
+                    result = true;
+                }
+                catch (Exception e)
+                {
+                    result = false;
+                }
+
+                if (module.Board.Connection.DelayAfterFirmwareUpdate > 0)
+                {
+                    System.Threading.Thread.Sleep(module.Board.Connection.DelayAfterFirmwareUpdate);
+                }
+            }
+            else
             {
                 Log.Instance.log($"Firmware update requested for {module.Board.Info.MobiFlightType} ({module.Port}) however no update settings were specified in the board definition file. Module update skipped.", LogSeverity.Warn);
             }
             return result;
         }
 
+        public static void FlashViaUsbDrive(Board board)

```

Comment:
The meat of the flashing. It's super simple! Check out the code comments for details on what's going on.